cmd/attest.go

```diff
@@ -0,0 +1,173 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"strconv"
+
+	"cloud.google.com/go/compute/metadata"
+	"github.com/google/go-tpm-tools/client"
+	"github.com/google/go-tpm-tools/proto/attest"
+	"github.com/google/go-tpm/tpm2"
+	"github.com/spf13/cobra"
+	"google.golang.org/protobuf/proto"
+)
+
+var (
+	key      string
+	teeNonce []byte
+)
+
+var attestationKeys = map[string]map[tpm2.Algorithm]func(rw io.ReadWriter) (*client.Key, error){
+	"AK": {
+		tpm2.AlgRSA: client.AttestationKeyRSA,
+		tpm2.AlgECC: client.AttestationKeyECC,
+	},
+	"gceAK": {
+		tpm2.AlgRSA: client.GceAttestationKeyRSA,
+		tpm2.AlgECC: client.GceAttestationKeyECC,
+	},
+}
+
+var attestCmd = &cobra.Command{
+	Use:   "attest",
+	Short: "Create a remote attestation report",
+	Long: `Gather information for remote attestation.
+The Attestation report contains a quote on all available PCR banks, a way to validate 
+the quote, and a TCG Event Log (Linux only).
+Use --key to specify the type of attestation key. It can be gceAK for GCE attestation
+key or AK for a custom attestation key. By default it uses AK.
+--algo flag override the public key algorithm for attestation key. If not provided then
+by default rsa is used.
+--nonce attaches even length extra data to the report and guarantees a fresh quote.
+--teenonce attaches a 64 bytes extra data to the attestation report of hardware and 
```

Comment:
64 bytes is just a coincidence that both TDX and SEV-SNP have the same length for their report_data field. A different TEE technology could have a different length.

Reply:
For now TDX and SEV-SNP have 64 byte report data feild. 
So we will add comments in cmd/attest.go stating 
" If hardware technology needs a variable length teenonce then please modify the flags description"